### PR TITLE
[Modal] Fix spacing in footer

### LINKED ISSUE #9265: Fix spacing for footer in Modal
## Issue summary

When using the `Modal` component, the footer padding should be consistent with the header and section padding. 

## Expected behavior

The `paddingInlineStart` and `paddingInlineEnd` in the `Footer` component should be "5", consistent with header and section padding. 

## Actual behavior

The `paddingInlineStart` and `paddingInlineEnd` in the `Footer` component is `"4"`

<details>
  <summary>Demo of issue</summary>
image
</details>

## Steps to reproduce the problem
1. Add a footer to the `Modal` component.

### PR DESCRIPTION
**Pull request by aishad**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/9265
When using the **Modal** component, the **Footer** padding should be consistent with the **Header** and **Section** padding.

### WHAT is this pull request doing?
Add `paddingInlineStart="5"` and `paddingInlineEnd="5"` to **Box** in **Footer** component to match padding for **Header** and **Section** in the **Modal** component.  

<details>
  <summary>Before</summary>
Before image with padding at
</details>

<details>
  <summary>After</summary>
Before image with padding at
</details>

### How to 🎩

[Spin link](https://shop1.shopify.footer-spacing.aisha-diallo.us.spin.dev/admin/customers): Open any modal, for example go to `admin/customers` and click **Import customers** to view the spacing in the footer of the modal.

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';

import {Page, Modal} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      <Modal
        open
        onClose={() => {}}
        title="Reach more shoppers with Instagram product tags"
        primaryAction={{
          content: 'Add Instagram',
          onAction: () => {},
        }}
        secondaryActions={[
          {
            content: 'Learn more',
            onAction: () => {},
          },
        ]}
        footer="Testing spacing in the footer."
      >
        <Modal.Section>
          <p>
            Use Instagram posts to share your products with millions of people.
            Let shoppers buy from your store without leaving Instagram.
          </p>
        </Modal.Section>
      </Modal>
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes (N/A)
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide (N/A)

### CODE REVIEW

File: polaris-react/src/components/Modal/tests/Modal.test.tsx [hunk 1]
```diff
@@ -341,6 +341,16 @@ describe('<Modal>', () => {
...
+
+      expect(modal).toContainReactComponent(Box, {
+        borderColor: 'border-subdued',
+        borderBlockStartWidth: '1',
+        minHeight: 'var(--p-space-16)',
+        padding: '4',
+        paddingInlineStart: '5',
+        paddingInlineEnd: '5',
+        width: '100%',
+      });
```
Comment (alex-page) [inline polaris-react/src/components/Modal/tests/Modal.test.tsx]: I wonder if this text is needed. It just tests that the properties on `Box` are the properties that are defined in the `Footer.tsx` file. I would be happy to ship it as is without this test.

**Review by alex-page — APPROVED** [review]

File: polaris-react/src/components/Modal/tests/Modal.test.tsx (hunk 1 above)
Comment (aishad) [reply polaris-react/src/components/Modal/tests/Modal.test.tsx]: Sure, wasn't sure if it was necessary to test this. Can remove it.
Thanks.

File: .changeset/witty-lions-camp.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+Fix paddingInlineStart and paddingInlineEnd on Box in Footer
```
Comment (kyledurand) [inline .changeset/witty-lions-camp.md]:
```suggestion
Fixed inline padding on Modal Footer
```

**Review by kyledurand — APPROVED** [review]

**Pull request merged**